thesis-api/main/persistence/category_dao.py

```diff
@@ -1,17 +1,16 @@
-from fastapi import Depends
+from sqlalchemy.orm import Session
 
-from database import SessionLocal, get_db
 from database.models import CategoryModel
 from persistence.objects import Category
 
 
 class CategoryDAO:
-    def __init__(self, db: SessionLocal = Depends(get_db)):
+    def __init__(self, db: Session):
         self.__db = db
 
-    def get_all(self) -> list[Category]:
-        categories = self.__db.query(CategoryModel).all()
+    def exists(self, category_id: int) -> bool:
+        return self.__db.query(CategoryModel).filter(CategoryModel.id == category_id).count() > 0
 
-        if categories is None:
-            return []
-        return [Category(id=c.id, name=c.name) for c in categories]
+    def get_all(self) -> list[Category]:
+        models = self.__db.query(CategoryModel).all() or []
```

Comment:
I'm not sure if or [] is necessary as the documentation says, all() returns the result of the query as a list. Therefore an empty result is returned as an empty list I guess ...